tests/integration/federation/test.sh

```diff
@@ -127,7 +127,16 @@ function show_workload_status() {
   fi
 
   echo "cofidectl workload status successful"
-  exit 0
+}
+
+function teardown_federation_and_verify() {
+  kubectl exec --context $K8S_CLUSTER_2_CONTEXT -n spire spire-server-0 -- /opt/spire/bin/spire-server federation delete -id td1
+  kubectl --context $K8S_CLUSTER_2_CONTEXT delete clusterspiffeids.spire.spiffe.io spire-spire-namespace
```

Comment:
Should this go first to avoid a potential race?

Reply:
Yeah I think this could be the culprit

Reply:
Re-run was successful, but potentially one to keep an eye on